frontend/components/ifixit-image/ifixit-image.tsx

```diff
@@ -1,19 +1,45 @@
-import Image, { ImageProps, ImageLoader } from 'next/image';
+import Image, { ImageProps, ImageLoader, ImageLoaderProps } from 'next/image';
+
+interface SizeMap {
+   [index: string]: number;
+}
+
+const guideImageSizeMap: SizeMap = {
+   mini: 56,
+   thumbnail: 96,
+   '140x105': 140,
+   '200x150': 200,
+   standard: 300,
+   '440x330': 440,
+   medium: 592,
+   large: 800,
+   huge: 1600,
+};
+
+const cartImageSizeMap: SizeMap = {
+   mini: 41,
+   thumbnail: 70,
+   size110: 110,
+   size170: 170,
+   size250: 250,
+   size400: 400,
+   medium: 600,
+   size1000: 1000,
+   large: 3000,
+};
 
 export function IfixitImage(props: ImageProps) {
-   let unoptimized = props.unoptimized;
+   let loader = props.loader;
 
    if (typeof props.src === 'string') {
       if (isGuideImage(props.src)) {
-         unoptimized = true;
+         loader = getImageLoader(guideImageSizeMap, 'huge');
       } else if (isCartImage(props.src)) {
-         unoptimized = true;
-      } else if (isStrapiImage(props.src)) {
-         unoptimized = true;
+         loader = getImageLoader(cartImageSizeMap, 'large');
       }
    }
 
-   return <Image {...props} unoptimized={unoptimized} />;
+   return <Image {...props} loader={loader} />;
```

Comment:
I'm not sure this is right for strapi. IIRC we use strapi's image resizing, so we should be able to take advantage of resized images.

Reply:
[9294d5c](https://github.com/iFixit/react-commerce/pull/537/commits/9294d5c5abf0d1bda0917c0b2814176828a270c8)